src/test/java/io/confluent/connect/elasticsearch/integration/SecurityIT.java

```diff
@@ -0,0 +1,119 @@
+      + "\"payload\": { \"key1\": 12, \"key2\": 15} }";
+  private static final String CONNECTOR_NAME = "elastic-sink";
+  private static final String KAFKA_TOPIC = "test-elasticsearch-sink";
+  private static final String TYPE_NAME = "kafka-connect";
+  private static final int TASKS_MAX = 1;
+  private static final int RETRY__VERIFY_SECONDS = 30;
+
+  @Before
+  public void setup() throws IOException {
+    connect = new EmbeddedConnectCluster.Builder().name("elastic-sink-cluster").build();
+    connect.start();
+  }
+
+  @After
+  public void close() {
+    connect.stop();
+  }
+
+  private Map<String, String> getProps () {
+    Map<String, String> props = new HashMap<>();
+    props.put(CONNECTOR_CLASS_CONFIG, ElasticsearchSinkConnector.class.getName());
+    props.put(SinkConnectorConfig.TOPICS_CONFIG, KAFKA_TOPIC);
+    props.put(TASKS_MAX_CONFIG, Integer.toString(TASKS_MAX));
+    props.put(VALUE_CONVERTER_CLASS_CONFIG, JsonConverter.class.getName());
+    props.put("type.name", TYPE_NAME);
+    props.put("batch.size", "1");
+    return props;
+  }
+
+  @Test
+  public void testSecureConnection() throws Throwable {
+    // TODO: Find a more robust way to get the IP address
+    final String address = "https://172.17.0.1:9200";
+
+    connect.kafka().createTopic(KAFKA_TOPIC, 1);
+
+    // Start connector
+    Map<String, String> props = getProps();
+    props.put("connection.url", address);
+    props.put("elastic.https.ssl.keystore.location", "./src/test/resources/certs/keystore.jks");
+    props.put("elastic.https.ssl.keystore.password", "asdfasdf");
+    props.put("elastic.https.ssl.key.password", "asdfasdf");
+    props.put("elastic.https.ssl.truststore.location", "./src/test/resources/certs/truststore.jks");
+    props.put("elastic.https.ssl.truststore.password", "asdfasdf");
+    connect.configureConnector(CONNECTOR_NAME, props);
+    waitForCondition(() -> {
+      ConnectorStateInfo info = connect.connectorStatus(CONNECTOR_NAME);
+      return info != null && info.tasks() != null && info.tasks().size() == 1;
+    }, "Timed out waiting for connector task to start");
+
+    connect.kafka().produce(KAFKA_TOPIC, MESSAGE_KEY, MESSAGE_VAL);
+
+    verify(getClient(props));
+  }
+
+  private JestClient getClient(Map<String, String> props) {
+    JestClientFactory factory = new JestClientFactory();
+    factory.setHttpClientConfig(getClientConfig(props));
+    return factory.getObject();
+  }
+
+  private void verify(JestClient client) throws Throwable {
+    // Read the message out of elastic directly via Jest
+    final Get get = new Get.Builder(KAFKA_TOPIC, MESSAGE_KEY).type(TYPE_NAME).build();
+
+    for (int i=0; i<RETRY__VERIFY_SECONDS; i++) {
+      try {
+        JsonObject result = client.execute(get).getJsonObject();
+        Boolean success = Boolean.parseBoolean(result.get("found").getAsString());
+        if (success) return;
+      } catch (Exception e) {
```

Comment:
also: this loop can be replaced with a `waitForCondition` as you have done above.